added Website Section paths builder

prceasar opened the pull request.

camiloriosf reviewed (approved):
LGTM!

tavurth commented on packages/lastrev-build-prefetch-content/src/buildTasks/writeWebsiteSectionPaths.ts:
```diff
@@ -0,0 +1,124 @@
+      let componentId;
+      let path = get(page, 'fields.slug');
+      let domain;
+      while (!domain) {
```
In my experience it's a little hard to understand which variables are strictly required for each level of a while loop.
To make this more maintainable we could try using a more functional approach:

```js
    const domain = parents
      // Find parentId and filter out falsy values
      .map(prop("sys.id"))
      .filter(identity)

      // Find the domain
      .find((parentId) => {
        const wss = get(websiteSections, parentId);
        if (!wss) return false;

        ... rest of logic
      });
```

This would make it easier for the next reader to determine what's happening at each level of the loop.

tavurth commented on packages/lastrev-build-prefetch-content/src/buildTasks/writeWebsiteSectionPaths.ts:
```diff
@@ -0,0 +1,124 @@
+      while (!domain) {
+        wss = get(websiteSections, parentId);
+        if (!wss) break;
+        if (!componentId) {
```
Based on my comment above, I believe this could be a bug which was overlooked because of the difficulty parsing the while loop.

`componentId` can be based on a different parent than the `domain` parent?

Loops:
1) `componentId` not found, extract and save `componentId` (null result), no domain found
2) `componentId` not found, extract and save `componentId` (truthy result), no domain found
3) `componentId` found in last step, (truthy result), domain found

Now `componentId` is coming from `step2` while `domain` is coming from `step3`.

If this is the intended logic it would be great to have a comment explaining why it works that way.

prceasar replied:
Yes, this is the intended behavior (Domain is always at the very top of the hierarchy, componentId comes from the lowest level wss of the hierarchy). I will add comments, and if possible try to refactor to a more functional approach

prceasar commented:
@tavurth @camiloriosf @maxtechera @spindle79 @bradtaylorsf I have made some updates to make this a little easier to understand, added comments, and added some tests. Please review and give me any additional, comments. Thanks!

tavurth commented:
> I have made some updates to make this a little easier to understand

Much cleaner, Thank you! 🎉
